dbt/adapters/spark/pysparkcon.py

```diff
@@ -0,0 +1,93 @@
+
+from __future__ import annotations
+
+import datetime as dt
+from types import TracebackType
+from typing import Any
+
+from dbt.events import AdapterLogger
+from dbt.utils import DECIMALS
+
+
+from pyspark.rdd import _load_from_socket
+import pyspark.sql.functions as F
+
+
+import importlib
+import sqlalchemy
+import re
+
+logger = AdapterLogger("Spark")
+NUMBERS = DECIMALS + (int, float)
+
+
+class PysparkConnectionWrapper(object):
+    """Wrap a Spark context"""
+
+    def __init__(self, python_module):
+        self.result = None
+        if python_module:
+            logger.debug(f"Loading spark context from python module {python_module}")
+            module = importlib.import_module(python_module)
+            create_spark_context = getattr(module, "create_spark_context")
+            self.spark = create_spark_context()
+        else:
+            # Create a default pyspark context
+            self.spark = SparkSession.builder.getOrCreate()
+
+    def cursor(self):
+        return self
+
+    def rollback(self, *args, **kwargs):
+        logger.debug("NotImplemented: rollback")
+
+    def fetchall(self):
+        try:
+            rows = self.result.collect()
+            logger.debug(rows)
+        except Exception as e:
+            logger.debug(f"raising error {e}")
+            dbt.exceptions.raise_database_error(e)
+        return rows
+
+    def execute(self, sql, bindings=None):
+        if sql.strip().endswith(";"):
+            sql = sql.strip()[:-1]
+
+        if bindings is not None:
+            bindings = [self._fix_binding(binding) for binding in bindings]
+            sql = sql % tuple(bindings)
+        logger.debug(f"execute sql:{sql}")
+        try:
+            self.result = self.spark.sql(sql)
+            logger.debug("Executed with no errors")
+            if "show tables" in sql:
+                self.result = self.result.withColumn("description", F.lit(""))
```

Comment:
why add the description column?

Reply:
this is an iceberg specific issue. When using iceberg it's missing the column. I'll remove this from the PR